src/db.rs

```diff
@@ -252,6 +263,12 @@ pub async fn data_products_by_dataset_select(
 }
 
 /// Upsert a new Dependency between Data Products
+// @coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
```

Comment:
@coderabbitai: Please generate a unit test that tests the following function for the following scenarios:
- Can we add a dependence?
- Can we update a dependence?
- Do we get rejected if the dependency parent refers to a non-existent data product?
- Do we get rejected if the dependency child refers to a non-existent data product?
- Do we get rejected if a dependency has the same data product for both the parent and child?